### PR TITLE
MbxUnderlineTabs

### LINKED ISSUE #9: MbxUnderlineTabs component
The details of this component are surprisingly difficult to get completely right, so I was thinking it could be a good first candidate for the `Mbx*` line of more opinionated components.

I'm of aware of 3 current variants that were deliberately styled:

screen shot 2018-07-27 at 10 26 18 am
screen shot 2018-07-27 at 10 26 21 am
screen shot 2018-07-27 at 10 26 25 am

Some variations in those patterns:

- Text size: `txt-xs`, `txt-s`, and regular.
- Whether inactive items are dimmed or not.
- Whether active items are bold or not.
- Color of the active-item underline.
- Distance between items (corresponds to text size).
- Height (corresponds to text size).
- Underline length — whether it exceeds the text by a couple of pixels or not.
- Horizontal overflow: the page-shell subnav tabs scroll horizontally when they run out of space; the larger docs tabs grow a "More" button with a dropdown.
- Hover state: only the small version in Studio has a hover state, darkening the dimmed text.

I'm leaning towards make the following choices (knowing that the Studio team is going to be best suited to maintaining their own variant):

- Don't dim the inactive tabs, so there's no danger of it looking disabled.
- Two sizes: small (current subnav) and large.
- Bold active items.
- Trim underline length to the text, so the component can be flush with the edge of its container.
- Active underline matches text color.
- Inactive items get blue text on hover (?).
- Build in the ability to transform some or all of the items into a dropdown.

@maikyon @angel I know you both worked on design for the two variants outside of Studio. Any opinions on the above?

### PR DESCRIPTION
**Pull request by davidtheclark**
Closes #9.

Here's what we have so far:

screen shot 2018-07-31 at 9 14 04 am

localhost_9966_mbxunderlinetabs

### CODE REVIEW

File: src/components/mbx-underline-tabs/mbx-underline-tabs.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+MbxUnderlineTabs.propTypes = {
+  /**
+   * Each item is an object with the following properties:
+   * - `id` (required): A string ID.
+   * - `label`: Text or React element. If `label` is not provided, `id` will
+   *   be used, with its first letter capitalized. (You should provide your
+   *   own `label` unless your `id`s are single words.)
+   * - `href`: A URL. If `href` is provided, the items will be `<a>`s.
+   *   If not, the items will be `<button>`s.
+   * -`disabled`: Boolean.
```
Comment (jseppi) [inline src/components/mbx-underline-tabs/mbx-underline-tabs.js]: I think you need a space between - and `
screen shot 2018-08-01 at 3 17 07 pm

File: src/components/mbx-underline-tabs/mbx-underline-tabs.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+      'txt-bold': props.bold,
+      'txt-s': medium,
+      'txt-xs': small
+    });
+
+    const itemEls = props.items.map((item, index) => {
+      const first = index === 0;
+      const layoutClasses = classnames('flex-child', {
+        ml12: !first && small,
+        'ml24 ml36-mxl': !first && medium,
```
Comment (jseppi) [inline src/components/mbx-underline-tabs/mbx-underline-tabs.js]: Checking that you want the medium list size to have more margin on xl screen sizes and that the same should not happen for the large list size.

File: src/components/mbx-underline-tabs/examples/mbx-underline-tabs-a.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
+/*
+Standard usage.
+*/
+import React from 'react';
+import MbxUnderlineTabs from '../mbx-underline-tabs';
+
+const items = [
```
Comment (jseppi) [inline src/components/mbx-underline-tabs/examples/mbx-underline-tabs-a.js]: Might be nice to include an item with the `href` property, for demonstration.

File: src/components/mbx-underline-tabs/__tests__/mbx-underline-tabs.test.js [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+  });
+
+  test('clicking an inactive item triggers change with the ID and event', () => {
+    const mockEvent = {};
+    wrapper.find({ 'data-test': 'A' }).prop('onClick')(mockEvent);
+    expect(testCase.props.onChange).toHaveBeenCalledTimes(1);
+    expect(testCase.props.onChange).toHaveBeenCalledWith('A', mockEvent);
+  });
+
+  test('clicking an active item does not trigger change and cancels the event', () => {
```
Comment (jseppi) [inline src/components/mbx-underline-tabs/__tests__/mbx-underline-tabs.test.js]: Perhaps add a similar test for a disabled item.

**Comment by davidtheclark** [comment]
Thanks for the great suggestions, @jseppi. I'll incorporate them all before merging.

**Pull request merged**